src/setup/cortx_deploy

```diff
@@ -94,13 +101,49 @@ validate_config $CLUSTER_INFO
 
 export PATH=$PATH:/opt/seagate/provisioner/bin
 
-# Apply Cluster Config
-log $INFO "cortx_setup config apply -f $(yaml_config $CLUSTER_INFO) -c '$CORTX_CONFSTORE'"
-cortx_setup config apply -f $(yaml_config $CLUSTER_INFO) -c "$CORTX_CONFSTORE";
+# Check system is pre-configured:
+# config apply: check confstore file.
+# cluster bootstrap: read overall_mp_status from state file.
 
-# Apply Common Config
-log $INFO "cortx_setup config apply -f $(yaml_config $CONFIG_INFO) -c '$CORTX_CONFSTORE'"
-cortx_setup config apply -f $(yaml_config $CONFIG_INFO) -c "$CORTX_CONFSTORE";
+
+FILE=$(awk -F:// '{print $2}' <<< $CORTX_CONFSTORE)
+if [ -f "$FILE" ]; then
+    log $INFO "File $FILE exists. System looks to be pre-configured."
+    CONFIG_APPLY=false
+fi
+local_path=`conf $CORTX_CONFSTORE get 'cortx>common>storage>local'`
+local_path=$(echo $local_path | tr -d "["\" | tr -d "\"]")
+node_id=`cat /etc/machine-id`
+STATE_FILE="$local_path/provisioner/state/status_$node_id"
```

Comment:
We should use conf store for storing node state so that information is available at one place.

Reply:
Done, 

At cluster_bootstrap stage, adding below info in conf_store and creating VERSION file only if deployment succeed.
`cortx:
  common:
   install_image: ghcr.io/seagate/cortx-all
   install_version: 2.0.0-1172-custom-ci`
   
  ```
 node:
    b1970cd0c8704a2fbaebf36419d27a19:
       phase: upgrade
       status: progress
       version: /opt/seagate/cortx/VERSION.INFO
       release: default
```